manager/state/raft/storage.go

```diff
@@ -229,6 +244,8 @@ func (n *Node) doSnapshot(ctx context.Context, raftConfig api.RaftConfig) {
 	// Wait for the goroutine to establish a read transaction, to make
 	// sure it sees the state as of this moment.
 	<-viewStarted
+
+	doSnapshotLatencyMetric.Observe(float64(time.Since(start).Nanoseconds()))
```

Comment:
This won't wait for the full snapshot process - that happens inside the goroutine.